Backend/editor/data/applicator.go

```diff
@@ -0,0 +1,44 @@
+package data
+
+// applicator manages the application of request objects to a provided datamodel
+
+import (
+	"errors"
+	"reflect"
+
+	"cms.csesoc.unsw.edu.au/editor/data/datamodels"
+)
+
+// TODO:
+// - Add error checking for the paths as we traverse, e.g missing an index when traversing an array (assuming we didn't reach the end)
+// - Make sure the item we are adding keeps the validity of the object
+
+// ApplyRequest takes a datamodel (as defined in the datamodels folder) and a request, it then proceeds to apply the request
+// to the model, note that this assumes that the operation in the request has been appropriately transformed
+func ApplyRequest(model datamodels.DataModel, request Request) error {
+	_, err := GetOperationTargetSite(model, request.path)
+	if err != nil {
```

Comment:
Could be 1 line, up to you :))

Reply:
This is temporary placeholder function, when actually implemented the ignored variable will be "unignored" so best to keep it as is